src/commands/detectorDiagnostics.ts

```diff
@@ -0,0 +1,152 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from "vscode-azureextensionui";
+import { AppLensARMresponse } from './models/detectorAppLens';
+import * as path from 'path';
+
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+export default async function detectorDiagnostics(
+    context: IActionContext,
+    target: any
+  ): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    if (cloudExplorer.available) {
+      const cloudTarget = await cloudExplorer.api.resolveCommandTarget(target);
+
+      if (cloudTarget && cloudTarget.cloudName === "Azure" &&
+            cloudTarget.nodeType === "resource" && cloudTarget.cloudResource.nodeType === "cluster") {
+            const armId = cloudTarget.cloudResource.armId;
+            // Get applens data
+            const clusterAppLensData = await getAppLensDetectorData(armId);
+            // Create webview
+            await CreateDetectorWebView(target.value.name, clusterAppLensData);
+        } else {
+          vscode.window.showInformationMessage('Please select kubernetes cluster.');
+        }
+    }
+}
+
+async function CreateDetectorWebView(
+  clusterName: string,
+  clusterAppLensData: (AppLensARMresponse | void)) {
+
+  if (clusterAppLensData) {
+    vscode.window.showInformationMessage("Loading...");
+    // Create webview
+    const panel = vscode.window.createWebviewPanel(
+      'AKS Diagnostics',
+      'AKS diagnostics view for: ' + clusterName,
+      vscode.ViewColumn.One,
+      {
+        enableScripts: true,
+        enableCommandUris: true
+      }
+    );
+
+    // Build webview from data.
+    panel.webview.html = getWebviewContent(clusterAppLensData);
+    vscode.window.showInformationMessage("AKS Diagnostics Completed.");
+  } else {
+    vscode.window.showInformationMessage('Selected Cluster has no data returned.');
+  }
+}
+
+async function getAppLensDetectorData(clusterARMId: string): Promise<AppLensARMresponse | void> {
+  // ARM api call.
+  const appLensARMAPI = `az rest -m GET -u "${clusterARMId}/detectors/mcrEndpointUpdate?api-version=2019-08-01&executeChildren=true"`;
+  const clusterAppLensData = await runAksShellCommand(appLensARMAPI);
+
+  return clusterAppLensData;
+}
+
+async function runAksShellCommand(azcomand: string): Promise<AppLensARMresponse | void> {
+  try {
+    const { stdout } = await exec(azcomand);
+    const appLensARMresponse = new AppLensARMresponse(JSON.parse(stdout));
+
+    return appLensARMresponse;
+  }catch (err) {
+    vscode.window.showInformationMessage('Error: ' + err);
+  }
+}
+
+function getWebviewContent(clusterdata: AppLensARMresponse) {
+    const webviewClusterData = clusterdata?.properties;
+    const vscodeExtensionPath = process.env.VSCODE_CWD;
```

Comment:
💡 `context.extensionPath` is only avalable level above, hence `process.env.VSCODE_CWD` is used, but I wonder if there is any other way?

Reply:
Pass in the `ExtensionContext`.  I'm not sure `VSCODE_CWD` is guaranteed stable - it's certainly not something I've seen before.

Reply:
💡 Cool, I found a way, seems like passing `extensioncontext` is not an option, I think the register command takes `IActionContext` **but** https://code.visualstudio.com/api/references/vscode-api#extensions.getExtension allows to get the extension context. `publisher`.`name` is accessible from `extension manifest`